frontstage/controllers/party_controller.py

```diff
@@ -715,13 +715,40 @@ def get_business_by_ru_ref(ru_ref: str):
     return response.json()
 
 
+def get_verification_token(party_id):
+    """
+    Gives call to party service to retrieve a verification token for the respondent
+    :param party_id: the respondent's id
+    :returns: verification token
+    """
+    logger.info("Attempting to retrieve respondent verification token", party_id=party_id)
+    url = f"{app.config['PARTY_URL']}/party-api/v1/respondents/{party_id}/password-verification-token"
+    response = requests.get(url, auth=app.config["BASIC_AUTH"])
+
+    try:
+        response.raise_for_status()
+    except requests.HTTPError:
+        if response == 404:
+            logger.error("Verification token not found")
+            raise NotFound("Token not found")
+        logger.error("Failed to retrieve verification token", party_id=party_id)
+        raise ApiError(logger, response)
+
+    logger.info("Successfully retrieved verification token")
+
+    return response.json()
+
+
 def post_verification_token(email, token):
     """
-    Gives call to party service to add a verification token for the respondent
+    Gives call to party service to add a verification token for the respondent and increase the password reset counter
     :param email: the respondent's email
     :param token: the verification token
     """
-    logger.info("Attempting to add respondent verification token", email=obfuscate_email(email))
+    logger.info(
+        "Attempting to add respondent verification token and increase password reset counter",
+        email=obfuscate_email(email),
+    )
 
     party_id = get_respondent_by_email(email)["id"]
     url = f"{app.config['PARTY_URL']}/party-api/v1/respondents/{party_id}/password-verification-tokens"
```

Comment:
wasn't `tokens` the deprecated one?  this (and all the other `-tokens` urls) should be changed to the new one to make less work for us later on